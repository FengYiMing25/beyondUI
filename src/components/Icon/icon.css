.icon {
  width: 1em;
  height: 1em;
}
.box {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  background: repeating-conic-gradient(#000 0, #000 0.8deg, transparent 1deg, transparent calc(360 / 60 * 1deg));
}
