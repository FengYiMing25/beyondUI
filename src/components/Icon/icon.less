.beyond-icon {
  width: 1.2em;
  height: 1.2em;
  display: inline-block;
  fill: currentColor; //加上这个可以给svg设置颜色
}
