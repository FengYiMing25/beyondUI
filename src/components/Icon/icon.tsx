import React, { FunctionComponent } from "react";
import "../../../lib/load-svg.js";
import "./icon.less";
import classNames from "../../util/classNames";
// 当添加新的svg图片是需要重新运行一下  yarn dev
interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
  title?: string;
}
const Icon: FunctionComponent<IconProps> = ({
  className,
  name,
  ...otherProps
}) => {
  return (
    <div>
      <svg className={classNames(["icon", className])} {...otherProps}>
        <use xlinkHref={`#${name}`}></use>
      </svg>
    </div>
  );
};
export default Icon;
