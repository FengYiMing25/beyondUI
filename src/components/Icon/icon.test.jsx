import * as TestRenderer from "react-test-renderer";
import React from "react";
import Icon from "./icon";
import {mount} from 'enzyme'
// 解决测试时候mount没有dom的情况
const jsdom = require("jsdom");
const { JSDOM } = jsdom;
const { window } = new JSDOM("");
const { document } = new JSDOM(``).window;
global.document = document;
global.window = window;

describe("icon", () => {
  // UI 测试
  it("UI 测试", () => {
    // 通过 toMatchSnapshot 形成一个快照来进行简单测试
    const json = TestRenderer.create(<Icon name="weChat" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  // 点击事件测试
  it("测试点击事件", () => {
    const fn = jest.fn()  
    const fn2 = jest.fn()
    const component = mount(<Icon name="happy" onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled()  //结果是fn被调用，fn2没有被调用
  });
});
