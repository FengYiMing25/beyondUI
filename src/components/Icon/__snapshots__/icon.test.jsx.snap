// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon UI 测试 1`] = `
<div>
  <svg
    className="icon"
  >
    <use
      xlinkHref="#weChat"
    />
  </svg>
</div>
`;
